internal/handlers/common/error.go

```diff
@@ -158,7 +171,89 @@ func formatBitwiseOperatorErr(err error, operator string, maskValue any) error {
 	}
 }
 
+// WriteErrors represents slice of protocol write errors.
+type WriteErrors []WriteError
+
+// Error implements error interface.
+func (we *WriteErrors) Error() string {
+	var err string
+	for _, e := range *we {
+		err += e.Error() + ","
+	}
+
+	return err
+}
+
+// Code returns error code.
+func (we *WriteErrors) Code() ErrorCode {
+	for _, e := range *we {
+		return e.code
+	}
+	return errUnset
+}
+
+// Unwrap implements standard error unwrapping interface.
+func (we *WriteErrors) Unwrap() error {
+	for _, e := range *we {
+		return &e
+	}
+	return nil
+}
+
+// Document returns wire protocol error document.
+func (we WriteErrors) Document() *types.Document {
+	errs := must.NotFail(types.NewArray())
+	for _, e := range we {
+		must.NoError(errs.Append(e.Document()))
+	}
+
+	d := must.NotFail(types.NewDocument(
+		"ok", float64(1),
+		"writeErrors", errs,
+	))
+	return d
+}
+
+// WriteError represents protocol write error.
```

Comment:
So is ProtoErr. We need better descriptions

Reply:
The second sentence explains why it exists.